lib/galaxy/webapps/galaxy/api/history_contents.py

```diff
@@ -874,7 +874,7 @@ def delete_legacy(
         self,
         response: Response,
         trans: ProvidesHistoryContext = DependsOnTrans,
-        history_id: DecodedDatabaseIdField = HistoryIDPathParam,
+        history_id: str = "",
```

Comment:
As a required Parameter it is misleading to provide a default value of an empty string, since you can't actually skip it for routing reasons. You can either move it one line higher or provide an annotation like this:
```suggestion
        history_id: str = Path(..., description="History ID or any string."),
```